charts/gsp-cluster/templates/02-gsp-system/egress-safelist.yaml

```diff
@@ -0,0 +1,78 @@
+{{- range $egressFqdn := .Values.egressSafelist }}
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: {{ $.Release.Name }}-egress-safelist-{{ $egressFqdn.name }}
+  namespace: {{ $.Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ include "gsp-cluster.name" $ }}-egress-safelist-{{ $egressFqdn.name }}
+    helm.sh/chart: {{ include "gsp-cluster.chart" $ }}
+    app.kubernetes.io/instance: {{ $.Release.Name }}
+    app.kubernetes.io/managed-by: {{ $.Release.Service }}
+spec:
+  hosts:
+  - {{ $egressFqdn.fqdn}}
+  ports:
+  - name: {{ $egressFqdn.name }}
+    number: {{ $egressFqdn.portNumber }}
+    protocol: {{ $egressFqdn.protocol }}
```

Comment:
I don't think this will work how you intend it do. You'll need to make this suite of templates protocol specific, as the structure of `VirtualService` and `DestinationRule` changes depending on whether it's HTTP, HTTPS, TLS or just TCP. So I would make this an "expose HTTP egress safelist" and set it up for that.

Reply:
I have reverted the changes back to the original change.